### ISSUE TITLE
Bad build of matplotlib on conda-forge channel?

### ISSUE THREAD

**Issue by ZLLentz** [conda-forge/matplotlib-feedstock#184]
Issue:

matplotlib=3.0.0 build 2 on conda-forge has no versioning attached to the pyqt dependency, despite not working with pyqt4. In some situation conda picks this version and pyqt4 by default, which breaks some CI builds for packages that depend on matplotlib. Using this build while explicitly picking pyqt=5 seems to work fine.

<details>

```
$ python
Python 3.6.7 |Anaconda, Inc.| (default, Oct 23 2018, 19:16:44) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import matplotlib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/__init__.py", line 1111, in <module>
    rcParamsOrig = RcParams(rcParams.copy())
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/__init__.py", line 891, in __getitem__
    from matplotlib import pyplot as plt
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/pyplot.py", line 32, in <module>
    import matplotlib.colorbar
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/colorbar.py", line 40, in <module>
    import matplotlib._constrained_layout as constrained_layout
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/_constrained_layout.py", line 52, in <module>
    from matplotlib.legend import Legend
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/legend.py", line 43, in <module>
    from matplotlib.offsetbox import HPacker, VPacker, TextArea, DrawingArea
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/offsetbox.py", line 33, in <module>
    from matplotlib.image import BboxImage
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/image.py", line 19, in <module>
    from matplotlib.backend_bases import FigureCanvasBase
  File "/u1/zlentz/conda/envs/mpl-test/lib/python3.6/site-packages/matplotlib/backend_bases.py", line 46, in <module>
    from matplotlib import (
ImportError: cannot import name 'get_backend'
```

</details>

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /u1/zlentz/conda/envs/mpl-test:
#
# Name                    Version                   Build  Channel
blas                      1.0                         mkl  
ca-certificates           2018.03.07                    0  
certifi                   2018.10.15               py36_0  
cycler                    0.10.0                   py36_0  
fontconfig                2.13.0               h9420a91_0  
freetype                  2.9.1                h8a8886c_1  
icu                       58.2                 h9c2bf20_1  
intel-openmp              2019.0                      118  
kiwisolver                1.0.1            py36hf484d3e_0  
libedit                   3.1.20170329         h6b74fdf_2  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 8.2.0                hdf63c60_1  
libgfortran-ng            7.3.0                hdf63c60_0  
libpng                    1.6.35               hbc83047_0  
libstdcxx-ng              8.2.0                hdf63c60_1  
libuuid                   1.0.3                h1bed415_2  
libxml2                   2.9.8                h26e45fe_1  
matplotlib                3.0.0                         2    conda-forge
matplotlib-base           3.0.0            py36hc039c98_2    conda-forge
mkl                       2019.0                      118  
mkl_fft                   1.0.6            py36h7dd41cf_0  
mkl_random                1.0.1            py36h4414c95_1  
ncurses                   6.1                  hf484d3e_0  
numpy                     1.15.3           py36h1d66e8a_0  
numpy-base                1.15.3           py36h81de0dd_0  
openssl                   1.1.1                h7b6447c_0  
pip                       10.0.1                   py36_0  
pyparsing                 2.2.2                    py36_0  
pyqt                      4.11.4                   py36_3    conda-forge
python                    3.6.7                h0371630_0  
python-dateutil           2.7.3                    py36_0  
qt                        4.8.7                         2  
readline                  7.0                  h7b6447c_5  
setuptools                40.4.3                   py36_0  
sip                       4.18                     py36_0  
six                       1.11.0                   py36_1  
sqlite                    3.25.2               h7b6447c_0  
tk                        8.6.8                hbc83047_0  
tornado                   5.1.1            py36h7b6447c_0  
wheel                     0.32.2                   py36_0  
xz                        5.2.4                h14c3975_4  
zlib                      1.2.11               ha838bed_2  
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
     active environment : mpl-test
    active env location : /u1/zlentz/conda/envs/mpl-test
            shell level : 2
       user config file : /u/lu/zlentz/.condarc
 populated config files : /u/lu/zlentz/.condarc
          conda version : 4.5.9
    conda-build version : 3.12.0
         python version : 3.6.6.final.0
       base environment : /u1/zlentz/conda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /u1/zlentz/conda/pkgs
                          /u/lu/zlentz/.conda/pkgs
       envs directories : /u1/zlentz/conda/envs
                          /u/lu/zlentz/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.9 requests/2.19.1 CPython/3.6.6 Linux/2.6.32-696.el6.x86_64 rhel/6.10 glibc/2.12
                UID:GID : 14059:1100
             netrc file : None
           offline mode : False


```
</details>

**Comment by ocefpaf** [comment]
It is pinned, see https://github.com/conda-forge/matplotlib-feedstock/blob/621b93d7d8c5d474c00141c8919ae6e07fa3704e/.ci_support/linux_python3.6.yaml#L34, but somehow that pin is not respected in the recipe with multiple outputs :-/

Here is the rendered recipe from that build:

```yaml
requirements:
    run:
        - matplotlib-base 3.0.0 py36hc039c98_2
        - pyqt
```

we should have the `5.6` in there. @msarahan any idea of what may be happening here?

**Comment by msarahan** [comment]
That is a run pin. Implicit pinning only happens in build and host.

**Comment by msarahan** [comment]
Note that run exports from hostc can and do apply pins to run. That is most commonly how implicit pins to a version in host end up as constraints in run.

**Comment by ocefpaf** [comment]
So we need to fix the recipe by adding it to host. @ZLLentz do you want to send a PR to fix it?

**Comment by ZLLentz** [comment]
Sure, just add `pyqt` to the `host` list in `meta.yaml`?

**Comment by msarahan** [comment]
Under that specific output, yes

**Comment by ZLLentz** [comment]
There are still builds on `conda-forge` with missing `pyqt` versioning. Are there plans to remove/replace these with corrected requirements? I don't think this issue should be closed as long as users can accidentally end up with these errored builds:
```
matplotlib 3.0.0 2
------------------
file name   : matplotlib-3.0.0-2.tar.bz2
name        : matplotlib
version     : 3.0.0
build string: 2
build number: 2
channel     : https://conda.anaconda.org/conda-forge/linux-64
size        : 5 KB
arch        : x86_64
constrains  : ()
license     : PSF-based
license_family: PSF
md5         : a2550a90b38ff5e361dbd4c933c9e2d1
platform    : linux
subdir      : linux-64
timestamp   : 1538577809921
url         : https://conda.anaconda.org/conda-forge/linux-64/matplotlib-3.0.0-2.tar.bz2
dependencies:
    matplotlib-base 3.0.0 py36hc039c98_2
    pyqt
```
```
matplotlib 3.0.1 0
------------------
file name   : matplotlib-3.0.1-0.tar.bz2
name        : matplotlib
version     : 3.0.1
build string: 0
build number: 0
channel     : https://conda.anaconda.org/conda-forge/linux-64
size        : 5 KB
arch        : x86_64
constrains  : ()
license     : PSF-based
license_family: PSF
md5         : 76856955c0d597663a9273587e7a9be8
platform    : linux
subdir      : linux-64
timestamp   : 1540493302212
url         : https://conda.anaconda.org/conda-forge/linux-64/matplotlib-3.0.1-0.tar.bz2
dependencies:
    matplotlib-base 3.0.1 py37hc039c98_0
    pyqt
```

**Comment by ocefpaf** [comment]
The issue was fixed for version `3.0.1` (build number `1`), so I moved the build number `0` to `broken`. However, we did not back-ported it to `3.0.0` and removing all `3.0.0` versions will removed the ability to install it, b/c `3.0.1` is a patch release that is not a big deal so I unless someone protest it here I'm removing that one too.
